js-packages/@prestojs/rest/src/viewModelCachingMiddleware.ts

```diff
@@ -208,6 +218,9 @@ export default function viewModelCachingMiddleware<ReturnT = any>(
                 if (!result) {
                     return {} as ReturnT;
                 }
+                if (!deleteViewModel) {
```

Comment:
is this condition necessary? previously if there was a result it would go through `cacheAndTransform` - it's just that it would use the same model as the delete occurred for

Reply:
This was required when I was looking at chaining. Now it's not necessarily required, but passing it through to cacheAndTransform seems a bit strange. If there is no deleteViewModel, then it's expecting either nothing or the response to be the model it just deleted. If it's the model you just deleted then it is kind of unexpected to delete it from the cache but then add it back in? Is there a case where that happens? Soft delete maybe?

Reply:
I guess you could return like a list of records, eg. if the deleted record changed ordering of other records

but I agree it's confusing and the previous behaviour wasn't really explicitly documented anyway. I'm happy with this - if you want response handled you have to specify `deleteViewModel`.

Can you update the documentation for the `viewModelMapping` to describe this, maybe:

```
* @param viewModelMapping The mapping to use for caching as described above. When the request method is `DELETE` this
 * is assumed to be the model that should be removed from the cache unless you provide `options.deleteViewModel`. If
 * `deleteViewModel` is provided then this value is used to cache the response of the delete call.
 ```

can you also add an example that shows deleting and handling the response?

Reply:
Done